File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<vector<int>> index;
        for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[0].size();j++){
                if(matrix[i][j]==0){
                    index.push_back({i,j});
                }
            }
        }
        for(auto k:index){
            for(int i=0;i<matrix.size();i++){
                matrix[i][k[1]]=0;
            }
            for(int i=0;i<matrix[0].size();i++){
                matrix[k[0]][i]=0;
            }
        }
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        List<int[]> zeros = new ArrayList<>();

        int rows = matrix.length;
        int cols = matrix[0].length;

        // Step 1: Record all zero positions
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                if (matrix[i][j] == 0) {
                    zeros.add(new int[]{i, j});
                }
            }
        }

        // Step 2: Set corresponding rows and columns to zero
        for (int[] pos : zeros) {
            int r = pos[0];
            int c = pos[1];

            for (int i = 0; i < cols; i++) matrix[r][i] = 0;  // Row
            for (int i = 0; i < rows; i++) matrix[i][c] = 0;  // Column
        }
    }
}